fmlloader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/ClasspathLocator.java

```diff
@@ -94,7 +94,7 @@ public void initArguments(Map<String, ?> arguments) {
     }
 
     private static boolean isValidManifest(SecureJar sj) {
-        try (var jis = new JarInputStream(Files.newInputStream(sj.getRootPath()))) {
+        try (var jis = new JarInputStream(Files.newInputStream(sj.getPrimaryPath()))) {
```

Comment:
This won't suffice for normal libraries.
The way CPL creates ModFiles requires a mods.toml to be available, so that they are loaded in the correct module.

See my other two PRs on this subject on how to fix that for the libraries.